### ISSUE TITLE
v4 <VisualEditing /> yields invalid hook call

### ISSUE THREAD

**Issue by dvnrsn** [sanity-io/hydrogen-sanity#72]
When I render `<VisualEditing />` in root we get the classic

```
Warning: Invalid hook call. Hooks can only be called inside of the body of a function component. This could happen for one of the following reasons:
1. You might have mismatching versions of React and the renderer (such as React DOM)
2. You might be breaking the Rules of Hooks
3. You might have more than one copy of React in the same app
See https://reactjs.org/link/invalid-hook-call for tips about how to debug and fix this problem.
TypeError: Cannot read properties of null (reading 'useSyncExternalStore')
```

node_modules/@sanity/visual-editing/node_modules/react/cjs/react.development.js:1676:21

```json
    "react": "^18.2.0",
    "hydrogen-sanity": "^4.0.2",

**Comment by nkgentile** [comment]
What does `npm ls react` output? (https://react.dev/warnings/invalid-hook-call-warning#duplicate-react)

If you see multiple versions of `react` (`18.2.x and 18.3.x`), then try running `npm dedupe`

**Comment by dvnrsn** [comment]
Are you on hydrogen v2024-04 and if so how do you avoid this issue?

https://github.com/Shopify/hydrogen/discussions/2073

**Comment by nkgentile** [comment]
Now that `react@18.3` has been released, you may want to use:
```json
   "react": "~18.2.0",
```

**Comment by dvnrsn** [comment]
approximate equivalence is not a bad idea but the point I am trying to make is that `@shopify/cli-hydrogen@8.0.0` has a dependency tree that includes as a peer dependency `react@"^18.3.1"`

Nonetheless, this change simply moves the hook error to the browser

```js
    "react": "~18.2.0",
    "react-dom": "~18.2.0",

**Comment by nkgentile** [comment]
When I encountered a similar error, I ran `npm ls react` and noticed both versions were installed. I then ran `npm dedupe` and they all resolved to `react@18.3.1`.

Another option might be to override the dependency? I haven't tested this though, so your mileage may vary
```
  "overrides": {
    "react": "~18.2"
  }
```

Since this isn't a dependency issue caused by this particular package, I'll close this issue.
